test/spec/provider/cloud-element-templates/ElementTemplates.spec.js

```diff
@@ -139,6 +139,157 @@ describe('provider/cloud-element-templates - ElementTemplates', function() {
 
   });
 
+  describe('getAll', function() {
+
+    it('should get all templates', inject(function(elementTemplates) {
+
+      // when
+      const templates = elementTemplates.getAll();
+
+      // then
+      expectTemplates(templates, [
+        [ 'my.mail.Task' ],
+        [ 'default', 1 ],
+        [ 'foo' ],
+        [ 'foo', 1 ],
+        [ 'foo', 2 ],
+        [ 'foo', 3 ],
+        [ 'bar', 1 ],
+        [ 'bar', 2 ],
+        [ 'baz' ]
+      ]);
+    }));
+
+
+    it('should get all template versions', inject(function(elementTemplates) {
+
+      // when
+      const templates = elementTemplates.getAll('foo');
+
+      // then
+      expectTemplates(templates, [
+        [ 'foo' ],
+        [ 'foo', 1 ],
+        [ 'foo', 2 ],
+        [ 'foo', 3 ],
+      ]);
+    }));
+
+
+    it('should get all applicable templates', inject(function(elementRegistry, elementTemplates) {
+
+      // given
+      const task = elementRegistry.get('Task_3');
+
+      // when
+      const templates = elementTemplates.getAll(task);
+
+      // then
+      expectTemplates(templates, [
+        [ 'foo' ],
+        [ 'foo', 1 ],
+        [ 'foo', 2 ],
+        [ 'foo', 3 ],
+        [ 'bar', 1 ],
+        [ 'bar', 2 ],
+        [ 'baz' ]
+      ]);
+    }));
+
+    it('should throw for invalid argument', inject(function(elementTemplates) {
+
+      // then
+      expect(function() {
+        elementTemplates.getAll(null);
+      }).to.throw('argument must be of type {String|djs.model.Base|Undefined}');
+
+    }));
+
+  });
+
+
+  describe('getLatest', function() {
+
+    it('should get all latest templates', inject(function(elementTemplates) {
+
+      // when
+      const templates = elementTemplates.getLatest();
+
+      // then
+      expectTemplates(templates, [
+        [ 'my.mail.Task' ],
+        [ 'default', 1 ],
+        [ 'foo', 3 ],
+        [ 'bar', 2 ],
+        [ 'baz' ]
+      ]);
+    }));
+
+
+    it('should get latest template version', inject(function(elementTemplates) {
+
+      // when
+      const templates = elementTemplates.getLatest('bar');
+
+      // then
+      expectTemplates(templates, [
+        [ 'bar', 2 ]
+      ]);
+    }));
+
+
+    it('should get latest template version (mixed versions)', inject(function(elementTemplates) {
+
+      // when
+      const templates = elementTemplates.getLatest('foo');
+
+      // then
+      expectTemplates(templates, [
+        [ 'foo', 3 ]
+      ]);
+    }));
```

Comment:
decided to go for "versioned templates are always newer than unversioned", as I see this as a possible step in the development process (first one does not have a version, second iteration does).